vxfreeswitch/tests/test_voice.py

```diff
@@ -560,12 +561,16 @@ def test_create_call(self):
         msg = self.tx_helper.make_outbound(
             'foobar', '12345', '54321', session_event='new')
 
-        client = yield self.esl_helper.mk_client(self.worker, 'uuid-1234')
-
         with LogCatcher(log_level=logging.WARN) as lc:
             yield self.tx_helper.dispatch_outbound(msg)
         self.assertEqual(lc.messages(), [])
 
+        client = yield self.esl_helper.mk_client(self.worker, 'uuid-1234')
+        # We need to wait for the client to be registered on the worker
+        d = defer.Deferred()
+        reactor.callLater(0.1, d.callback, None)
```

Comment:
I don't think there is a clean way to do this because we need to wait for an internal event inside the transport. One way to do this would be to write a small helper function that gets that the client id appears in `_unanswered_calls` and then waits a bit for it to disappear from `_unanswered_calls` (this last part would use a small loop trying, e.g., 3 times and using reactor.callLater to wait between each).